cai2hcl/common.go

```diff
@@ -3,23 +3,22 @@ package cai2hcl
 import (
 	"github.com/GoogleCloudPlatform/terraform-google-conversion/v2/caiasset"
 
-	tfschema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/zclconf/go-cty/cty"
 )
 
-// ConvertFunc converts an Asset type into a map[string]interface{} struct.
-type ConvertFunc func(asset *caiasset.Asset) (string, map[string]interface{}, error)
-
-// Converter for individual resource.
-type Converter struct {
-	TFResourceName string
-	Convert        ConvertFunc
-	Resource       *tfschema.Resource
+// Converter interface for resources.
+type Converter interface {
+	// Convert converts an Asset into ID, and cty.Value.
+	Convert(asset ...*caiasset.Asset) (id string, data cty.Value, iam cty.Value, err error)
```

Comment:
I like the idea of having the convert function return a cty.Value - that makes it a lot easier to customize the conversion per-resource if needed.